blocks/media/media.js

```diff
@@ -0,0 +1,275 @@
+  return `${SITE_ROOT}/research/videos/${permLink}`;
+}
+
+function getPodcastsShareLink(permLink) {
+  return `${SITE_ROOT}/research/podcasts/${permLink}`;
+}
+
+function getVideosThumbnail(id) {
+  return ICICI_DIRECT_VIDEOS_THUMBNAIL.replace('{id}', id);
+}
+
+function getPodcastsThumbnail(image) {
+  return `${SITE_ROOT}/images/${image}`;
+}
+
+function numberOfDots(totalCards, maxAllowedCards) {
+  if (totalCards > maxAllowedCards) return totalCards - maxAllowedCards + 1;
+  return 0;
+}
+
+function openUrl(event) {
+  window.location.href = event.currentTarget.getAttribute('share-link');
+}
+
+function allowedCardsCount() {
+  const deviceType = Viewport.getDeviceType();
+  switch (deviceType) {
+    case 'Desktop':
+      return 3;
+    case 'Tablet':
+      return 2;
+    default:
+      return 1;
+  }
+}
+
+function createCardMetaElement(...cardMetaValues) {
+  const postMeta = document.createElement('div');
+  postMeta.classList.add('card-meta');
+  // Iterate through the values
+  cardMetaValues.forEach((value) => {
+    const abbr = document.createElement('abbr');
+    abbr.textContent = value;
+    postMeta.appendChild(abbr);
+  });
+  return postMeta;
+}
+
+function createPostTitle(title, shareLink) {
+  const postTitle = document.createElement('h3');
+  postTitle.classList.add('card-title');
+  const postLink = document.createElement('a');
+  postLink.setAttribute('href', shareLink);
+  postLink.setAttribute('target', '_blank');
+  postLink.textContent = title;
+  postTitle.appendChild(postLink);
+  return postTitle;
+}
+
+function createSocialLinkElement(shareLink) {
+  const socialLink = document.createElement('div');
+  socialLink.classList.add('social-link');
+  const socialAnchor = document.createElement('a');
+  const socialIcon = document.createElement('i');
+  socialIcon.classList.add('fa', 'fa-share', 'icon');
+  socialAnchor.dataset.href = shareLink;
+  socialAnchor.addEventListener('click', () => handleSocialShareClick(socialAnchor));
+  socialAnchor.appendChild(socialIcon);
+  socialLink.appendChild(socialAnchor);
+  return socialLink;
+}
+
+function targetedDotView(event) {
+  const targetDotIndex = event.currentTarget.dataset.index;
+  const block = event.currentTarget.closest('.block.media');
+  clearInterval(block.getAttribute('interval-id'));
+  const track = block.querySelector('.track');
+  const dotsContainer = block.querySelector('.dots-container');
+  dotsContainer.querySelector('.active').classList.remove('active');
+  event.currentTarget.classList.add('active');
+  const cardSize = track.firstElementChild.offsetWidth;
+  const moveDistance = (targetDotIndex - 1) * cardSize;
+  track.style.transform = `translateX(-${moveDistance}px)`;
+}
+
+async function createPicture(imageUrl, mediaWrapper) {
+  mediaWrapper.appendChild(createPictureElement(imageUrl, 'mqdefault', false));
+}
+
+function createMediaCards(container, data, apiKey, cardWidth) {
+  data.forEach((item) => {
```

Comment:
suggestion , we can https://github.com/aemsites/icicidirect/blob/main/scripts/dom-builder.js here